vm/vm.go

```diff
@@ -95,6 +97,12 @@ func JunoAppendActualFee(readerHandle C.uintptr_t, ptr unsafe.Pointer) {
 	context.actualFees = append(context.actualFees, makeFeltFromPtr(ptr))
 }
 
+//export JunoAppendRevert
+func JunoAppendRevert(readerHandle C.uintptr_t, str *C.char) {
+	context := unwrapContext(readerHandle)
+	context.reverts = append(context.reverts, C.GoString(str))
+}
+
```

Comment:
Revert reasons are already available within the trace, so returning them separately is a bit overkill. Instead of returning the reason as a whole, we can get away with returning just the transaction indexes that are failing.